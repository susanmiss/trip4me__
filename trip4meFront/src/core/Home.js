import React from "react";
import Posts from '../post/Posts'
import Search from '../core/Search';
import Categories from '../category/Categories';


const Home = () => (
  <div>
    <header className="masthead">
      <div className="container">
        <div className="intro-text">
          <div className="intro-lead-in"></div>
          <div className="intro-heading text-uppercase"></div>
        </div>
      </div>
    </header>

    <div className="page-section bg-welcome">
      <div className="container">
        <div className="row">
          <div className="col-lg-12 text-center">
            <h2 className="section-heading">Welcome to Trip4me</h2>
            <h3 className="section-subheading text-welcome">'It's a dangerous business, Frodo, going out your door. You step onto the road, and if you don't keep your feet, there's no knowing where you might be swept off to...' Bilbo Baggings</h3>

          </div>
        </div>
      </div>
    </div>


    <div>
      <Search />
      <Categories />
    </div>

  </div>



);


export default Home;
