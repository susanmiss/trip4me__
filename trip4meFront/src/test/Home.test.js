import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme'; 
import renderer from 'react-test-renderer';
import Home from '../core/Home'


const setup = (props={}, state=null) => {
    const wrapper = shallow(<App {...props} />)
    if( state ) wrapper.setState(state);
    return wrapper;
  }

  const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  }

describe("<Home />", () => {

  it('mounts to DOM', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Home />, div);
      ReactDOM.unmountComponentAtNode(div);
  });
  it('renders correctly', () => {
      const tree = renderer
          .create(<Home />)
          .toJSON();
      expect(tree).toMatchSnapshot();
  });
  let wrapper;
  beforeEach(() => wrapper = shallow(<Home />));
  
  it('Renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  })
  
  it('should render a <div />', () => { 
      expect(wrapper.find('div').length).toEqual(10);
  });

    test('renders <Post/> component', () => {
    const postComponent = wrapper.find("[data-test='component-post']")
    expect(postComponent.length).toBe(1);
    })
 
})