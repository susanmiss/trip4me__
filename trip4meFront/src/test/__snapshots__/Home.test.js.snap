// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Renders correctly 1`] = `ShallowWrapper {}`;

exports[`<Home /> renders correctly 1`] = `
<div>
  <header
    className="masthead"
  >
    <div
      className="container"
    >
      <div
        className="intro-text"
      >
        <div
          className="intro-lead-in"
        />
        <div
          className="intro-heading text-uppercase"
        />
      </div>
    </div>
  </header>
  <div
    className="page-section bg-welcome"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12 text-center"
        >
          <h2
            className="section-heading text-uppercase"
          >
            Welcome to Trip4me
          </h2>
          <h3
            className="section-subheading text-welcome"
          >
            'It's a dangerous business, Frodo, going out your door. You step onto the road, and if you don't keep your feet, there's no knowing where you might be swept off to...' Bilbo Baggings
          </h3>
        </div>
      </div>
    </div>
  </div>
  <div>
    <section
      className="page-section"
      id="portfolio"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <h2
            className="mt-2"
          >
            No posts yet!
          </h2>
          <div
            className="container"
          >
            
            
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
